pkg/updatechecker/updatechecker.go

```diff
@@ -469,33 +469,40 @@ func autoDeploy(opts CheckForUpdatesOpts, clusterID string, semverAutoDeploy app
 
 	var versionToDeploy *downstreamtypes.DownstreamVersion
 
-Loop:
-	for _, v := range appVersions.AllVersions {
-		if v == nil || v.Semver == nil {
-			continue
-		}
+	// In the case semver is not required, we only need to check if the newest app version
+	// is greater than the current version.
+	if autoDeploy == apptypes.AutoDeploySequence && currentVersion.Sequence < appVersions.AllVersions[0].Sequence {
```

Comment:
what if `autoDeploy` is set to `apptypes.AutoDeploySequence` but the current sequence is greater than or equal to the latest version sequence? then it's gonna execute the semantic versioning logic below instead, which i think is incorrect.

Reply:
Yeah, great callout.